lib/background_streamer/worker.rb

```diff
@@ -3,142 +3,111 @@
 
 module BackgroundStreamer
   class Worker
-    CRLF= "\r\n".freeze
+    CRLF = "\r\n".freeze
+    RACK_HIJACK = 'rack.hijack'.freeze
+    RACK_HIJACK_IO = 'rack.hijack_io'.freeze
+    ACTION_DISPATCH_REQUEST_ID = 'action_dispatch.request_id'.freeze
+    X_REQUEST_ID = 'X_REQUEST_ID'.freeze
 
-    @count = 0
-    @middleware = []
+    attr_reader :options, :timeout, :request_id, :status, :headers, :io, :body
 
     class << self
-      attr_accessor :count
-      attr_accessor :middleware
-      def inc
-        @count += 1
-      end
+      def perform_async(env, body, options={})
+        env[RACK_HIJACK].call
+
+        Thread.new do 
```

Comment:
@ismell @mhuggins Ah, that's a good idea, green threads for greener pastures.

@ismell You get a trophy bird.